# SPDX-FileCopyrightText: (C) 2026 openRuyi Project Contributors
# SPDX-License-Identifier: MulanPSL-2.0
# SPDX-FileContributor: A.Donz <avrova.minix@donz.si>
# SPDX-FileContributor: MMLA-org <https://github.com/MMLA-org>
Upstream-Status: Downstream
Upstream-Reference: https://github.com/pytorch/vision/tree/v0.29.0

From c380725c6bf685a66155cb351b26c8e543a65ad9 Mon Sep 17 00:00:00 2001
From: CHEN Xuan <chenxuan@iscas.ac.cn>
Date: Thu, 20 Aug 2026 18:49:42 +0800
Subject: [PATCH] Add HIP version detection

---
 torchvision/csrc/vision_stable.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/torchvision/csrc/vision_stable.cpp b/torchvision/csrc/vision_stable.cpp
index 6cd289d..f834126 100644
--- a/torchvision/csrc/vision_stable.cpp
+++ b/torchvision/csrc/vision_stable.cpp
@@ -14,6 +14,7 @@
 #endif
 #ifdef WITH_HIP
 #include <hip/hip_runtime.h>
+#include <rocm-core/rocm_version.h>
 #endif

 // If we are in a Windows environment, we need to define
@@ -29,6 +30,8 @@ int64_t cuda_version() {
 #ifdef WITH_CUDA
   // looks like 12060, 12080, etc.
   return CUDA_VERSION;
+#elif defined(WITH_HIP)
+  return (ROCM_VERSION_MAJOR * 100) + ROCM_VERSION_MINOR;
 #else
   return -1;
 #endif
--
2.53.0
